cypress/e2e/article.cy.js

```diff
@@ -1,24 +1,123 @@
   });
 
-  it('should be created using New Article form', () => {
+  it(`should allow to edit an article using Edit button`, function () {
+    const { username } = this.userData;
+
+    articlePage.assertBannerContainsEditArticleBtn(btnNames.editArticle);
+    articlePage.clickOnEditArticleBtn();
+
+    cy.get('@articleSlug').then((slug) => {
+      const articleEditor = new ArticleEditorPageObject(slug);
+
+      articleEditor.assertPageUrl(articleEditor.url);
+    });
+
+    articleEditor.assertArticleFormContainsArticleParts(this.articleData);
+
+    cy.intercept('PUT', 'articles').as('articleEditing');
+    articleEditor.fillFormAndSubmit(this.newArticleData, ` edited`);
+
+    cy.wait('@articleEditing').then((interception) => {
+      const {
+        slug: updatedSlug,
+        title: updatedTitle,
+        body: updatedBody,
+        description: updatedDescription
+      } = interception.response.body.article;
 
+      const articlePage = new ArticlePageObject(updatedSlug);
+
+      articlePage.assertPageUrl(articlePage.url);
+      articlePage.assertArticleTitleUpdated(updatedTitle);
+      articlePage.assertArticleBodyUpdated(updatedBody);
+      articlePage.clickOnUsernameLink();
+
+      profilePage.assertUsernameExists(username);
+      profilePage
+        .assertArticlePreviewIsUpdated(updatedTitle, updatedDescription);
+    });
   });
 
-  it('should be edited using Edit button', () => {
+  it(`should allow to completely rewrite an article using Edit button`, function () {
+    const { username } = this.userData;
+
+    const {
+      title: updatedTitle,
+      body: updatedBody
+    } = this.newArticleData;
+
+    articlePage.clickOnEditArticleBtn();
+
+    articleEditor.assertArticleFormExists();
+
+    cy.intercept('PUT', 'articles').as('articleEditing');
+    articleEditor.editFormAndSubmit(this.newArticleData);
 
+    cy.wait('@articleEditing').then((interception) => {
+      const {
+        slug: updatedSlug
+      } = interception.response.body.article;
+
+      const articlePage = new ArticlePageObject(updatedSlug);
+
+      articlePage.assertPageUrl(articlePage.url);
+      articlePage.assertArticleTitleUpdated(updatedTitle);
+      articlePage.assertArticleBodyUpdated(updatedBody);
+      articlePage.clickOnUsernameLink();
+
+      profilePage.assertUsernameExists(username);
+      profilePage
+        .assertArticlePreviewIsUpdated(this.newArticleData);
+    });
   });
 
-  it('should be deleted using Delete button', () => {
+  it(`should be deleted using Delete button`, function () {
+    const { username } = this.userData;
+
+    articlePage.assertBannerContainsDeleteArticleBtn(btnNames.deleteArticle);
+    articlePage.clickOnDeleteArticleBtn();
+
+    homePage.assertPageUrl(homePage.url);
+    homePage.assertNoArticleAvailable(articleTip.noArticle);
+    homePage.clickOnUsernameLink();
 
+    profilePage.assertUsernameExists(username);
+    profilePage.assertNoArticleAvailable(articleTip.noArticle);
```

Comment:
I made this assertion because when the article is created the article preview becomes available on the 'Profile' page. And I think it is not redundant to make this assertion to make sure that the article is completely deleted and the user doesn't have access to it from the 'Profile' page